graphql_compiler/tests/snapshot_tests/test_cost_estimation.py

```diff
@@ -1324,87 +1323,87 @@ class IntegerIntervalTests(unittest.TestCase):
 
     def test_interval_creation(self) -> None:
```

Comment:
Some non-int tests on `intersect_intervals` would be nice, especially with date/datetime types.

Reply:
Restricted the domain of intersection to ints to avoid getting too out of scope for auto-pagination.